pytest failure because it uses ALLOWED_HOSTS from development .env

I have the following setting in my `.env` file:
```
## Leave unset to allow all hosts
ALLOWED_HOSTS="localhost,127.0.0.1,[::1]"
```

This causes pytest to fail on `test_signing.py`. When I comment out `ALLOWED_HOSTS`, the test succeeds. I think pytest should not look at the `.env` file, but use a different set of settings specifically for testing.

Failed test:
```
❯ ./bw-dev pytest bookwyrm/tests/test_signing.py
+ case "$CMD" in
+ prod_error
+ '[' true '!=' true ']'
+ runweb pytest --no-cov-on-fail bookwyrm/tests/test_signing.py
+ docker compose run --rm web pytest --no-cov-on-fail bookwyrm/tests/test_signing.py
[+] Creating 4/0
 ✔ Container bookwyrm-redis_broker-1    Running                                                                                                                                                                                                                                       0.0s
 ✔ Container bookwyrm-redis_activity-1  Running                                                                                                                                                                                                                                       0.0s
 ✔ Container bookwyrm-db-1              Running                                                                                                                                                                                                                                       0.0s
 ✔ Container bookwyrm-celery_worker-1   Running                                                                                                                                                                                                                                       0.0s
=================================================================================================================================== test session starts ===================================================================================================================================
platform linux -- Python 3.9.18, pytest-6.1.2, py-1.11.0, pluggy-0.13.1
django: settings: bookwyrm.settings (from ini)
rootdir: /app, configfile: pytest.ini
plugins: xdist-2.3.0, forked-1.6.0, django-4.1.0, cov-2.10.1, env-0.6.2
collected 8 items

bookwyrm/tests/test_signing.py FFFFFFFF                                                                                                                                                                                                                                             [100%]

======================================================================================================================================== FAILURES =========================================================================================================================================
_______________________________________________________________________________________________________________________________ Signature.test_changed_data _______________________________________________________________________________________________________________________________

self = <bookwyrm.tests.test_signing.Signature testMethod=test_changed_data>

    @pytest.mark.integration
    def test_changed_data(self):
        """Message data must match the digest header."""
        with patch("bookwyrm.activitypub.resolve_remote_id"):
            response = self.send_test_request(
                self.mouse, send_data=get_follow_activity(self.mouse, self.cat)
            )
>           self.assertEqual(response.status_code, 401)
E           AssertionError: 400 != 401

bookwyrm/tests/test_signing.py:205: AssertionError
---------------------------------------------------------------------------------------------------------------------------------- Captured stderr call -----------------------------------------------------------------------------------------------------------------------------------
Invalid HTTP_HOST header: 'your.domain.here'. You may need to add 'your.domain.here' to ALLOWED_HOSTS.
Bad Request: /user/rat/inbox
------------------------------------------------------------------------------------------------------------------------------------ Captured log call ------------------------------------------------------------------------------------------------------------------------------------
ERROR    django.security.DisallowedHost:exception.py:99 Invalid HTTP_HOST header: 'your.domain.here'. You may need to add 'your.domain.here' to ALLOWED_HOSTS.
WARNING  django.request:log.py:224 Bad Request: /user/rat/inbox

<...snip...>

================================================================================================================================= short test summary info =================================================================================================================================
FAILED bookwyrm/tests/test_signing.py::Signature::test_changed_data - AssertionError: 400 != 401
FAILED bookwyrm/tests/test_signing.py::Signature::test_correct_signature - AssertionError: 400 != 200
FAILED bookwyrm/tests/test_signing.py::Signature::test_invalid_digest - AssertionError: 400 != 401
FAILED bookwyrm/tests/test_signing.py::Signature::test_key_needs_refresh - AssertionError: 400 != 200
FAILED bookwyrm/tests/test_signing.py::Signature::test_nonexistent_signer - AssertionError: 400 != 401
FAILED bookwyrm/tests/test_signing.py::Signature::test_old_message - AssertionError: 400 != 401
FAILED bookwyrm/tests/test_signing.py::Signature::test_remote_signer - AssertionError: 400 != 200
FAILED bookwyrm/tests/test_signing.py::Signature::test_wrong_signature - AssertionError: 400 != 401
=================================================================================================================================== 8 failed in 20.37s ====================================================================================================================================
```

--- a/pytest.ini
+++ b/pytest.ini
@@ -11,6 +11,7 @@ env =
     DEBUG = false
     USE_HTTPS = true
     DOMAIN = your.domain.here
+    ALLOWED_HOSTS = your.domain.here
     BOOKWYRM_DATABASE_BACKEND = postgres
     MEDIA_ROOT = images/
     CELERY_BROKER = ""

